codec/packer.go

```diff
@@ -139,6 +140,32 @@ func (p *Packer) UnpackInt(required bool) int {
 	return int(v)
 }
 
+func (p *Packer) PackUvarInt(v uint64) {
+	p.p.Bytes = binary.AppendUvarint(p.p.Bytes, v)
```

Comment:
This does not properly update `p.Offset`: https://github.com/ava-labs/avalanchego/blob/ddf66eaed1659c2caa8531bbda83b360ca85900e/utils/wrappers/packing.go#L50

I think you should make a PR to AvalancheGo that adds support for packing `Uvarint` and `Varint`. We can then use that downstream here (doing it outside of the upstream library seems wrong given we are modifying its internal data structures).

Reply:
I have opened https://github.com/ava-labs/avalanchego/pull/3081 in order to do that downstream first, after that we will be able to call these methods directly !